WMF Framework/Remote Notifications/RemoteNotificationsOperationsController.swift

```diff
@@ -39,6 +39,18 @@ class RemoteNotificationsOperationsController: NSObject {
     deinit {
         NotificationCenter.default.removeObserver(self)
     }
+    
+    func deleteOldDatabaseFiles() throws {
+        let modelName = RemoteNotificationsModelController.modelName
+        let sharedAppContainerURL = FileManager.default.wmf_containerURL()
+        let legacyRemoteNotificationsStorageUrl = sharedAppContainerURL.appendingPathComponent(modelName)
+        let legecyJournalShmUrl = sharedAppContainerURL.appendingPathComponent("\(modelName)-shm")
+        let legecyJournalWalUrl = sharedAppContainerURL.appendingPathComponent("\(modelName)-wal")
+        
+        try FileManager.default.removeItem(at: legacyRemoteNotificationsStorageUrl)
```

Comment:
I think the preferred way to delete a persistent store is via [destroyPersistentStore](https://developer.apple.com/documentation/coredata/nspersistentstorecoordinator/1468888-destroypersistentstore), though it looks like some folks are reporting this isn't also deleting the meta optimization files for them.

Frankly, I can't personally recall deleting an in-use Core Data store and in place recreating it myself, so I don't have good guidance to offer here. But maybe, because of folks reporting some of their issues with the meta files not also being deleted, and the fact we're intending to rewrite a store in place with a very similar name, your approach seems like the more reasonable approach.

Reply:
@staykids thanks for catching this! I would certainly rather do it the official way, I didn't realize this method existed. I'll explore this a bit today.

Reply:
@staykids  I pushed a commit calling `destroyPersistentStore` in the migration method, before deleting the old files. It's maybe unnecessary, but I found that `destroyPersistentStore` alone didn't delete the files, and to gain that `sqlite` extension I would rather delete the old files and redo the stack than rename. `destroyPersistentStore` is here really in case it does any extra cleanup internally that I wouldn't know to do. I'll merge EOD or pause that if you have any concerns here. Thanks!